ocean_lib/models/data_nft.py

```diff
@@ -98,7 +101,9 @@ def get_default_token_uri(self):
             json.dumps(data, separators=(",", ":")).encode("utf-8")
         )
 
-    def deploy_contract(self, config_dict, tx_dict) -> DataNFT:
+    def deploy_contract(
+        self, config_dict, tx_dict
+    ) -> Union[DataNFT, TransactionReceipt]:
```

Comment:
I don't think you should modify these. I think we should throw an exception if the transaction fails